### ISSUE TITLE
Unable to add Integration (no territory available)

### ISSUE THREAD

**Issue by martinbeno** [dotKrad/hass-fpl#37]
1. I did download propper set of files info custom_components folder
2. I did restart HA
3. Then I was able to find FPL in addons
4. After filling email and password - the same I use to log into FPL portal ( https://www.fpl.com/my-account/login.html ) - I receive error msg "An error ocurred, please check the logs" and cannot continue.

Note: 

**Log file says:**
_2022-10-29 16:13:03.588 ERROR (MainThread) [custom_components.fpl] Error : <class 'custom_components.fpl.fplapi.NoTerrytoryAvailableException'>_

![image](https://user-images.githubusercontent.com/116977872/198851005-d5a630a5-c979-4e68-92d4-f84b50c86eda.png)

**Running:**
Home Assistant 2022.10.5
Supervisor 2022.10.0
Operating System 9.3

(fresh installation today)

**Comment by kamikazzi1** [comment]
I seem to have the same problem. Were you able to find a resolution to this?

**Comment by dotKrad** [comment]
May I ask what FPL you using, Northwest or regular one

**Comment by dotKrad** [comment]
> 

The thing is when I added the NW implementation I also added a region check for both regular and NW

**Comment by martinbeno** [comment]
> May I ask what FPL you using, Northwest or regular one

Regular one

**Comment by kamikazzi1** [comment]
Same issue here. Were you ever able to get this going? 
Error log states: "

Log Details (ERROR)
This error originated from a custom integration.
Logger: custom_components.fpl
Source: custom_components/fpl/fplapi.py:122
Integration: FPL ([documentation](https://github.com/dotKrad/hass-fpl), [issues](https://github.com/dotKrad/hass-fpl/issues))
First occurred: 6:47:45 AM (1 occurrences)
Last logged: 6:47:45 AM

Error : <class 'asyncio.exceptions.TimeoutError'>
"

**Comment by martinbeno** [comment]
Nope, went different way, installed Emporia Energy Monitor.

**Comment by dotKrad** [comment]
I'm sorry it isn't working for you guys, I keep testing on my end, but I get results, I have two accounts, ( one is mine and one from a contributor ) and both works, one is for main region and the other one for NW

**Comment by martinbeno** [comment]
Just to make sure, am I supposed to use the same login credential I use for FPL mobile app, or web site?

**Comment by dotKrad** [comment]
yes

**Comment by zoopster** [comment]
I wonder if this continues to be an issue for @martinbeno or @kamikazzi1 ? 
If you login to the website and then change the URL to `https://fpl.com/api/resources/account` what is returned?

**Comment by martinbeno** [comment]
Hello John,

at the end of a day, I decided to go different way and I measure my consumption myself, so I haven’t tried this addon since last year’s unsuccessful attempts.

Anyways, if it would be helpful for debugging for others, I’m willing to provide you with the JSON response I get. It’s quite big and as I could notice at first glance, it includes a lot of sensitive information I don’t feel comfortable to post here publicly. But, if you would provide me with your personal email address, I will be more than happy to send it to you.

Now I use Emporia Vue (quite nice piece of hardware and cheap), I didn’t use the integration  (https://github.com/magico13/ha-emporia-vue) for HA as it was not transparent for me and I don’t want to use HACS. I just used this JAVA downloader (https://github.com/helgew/emporia-downloader), placed it on one of my Linux servers I have connected to the Internet and wrote PHP script, that handles it all, and just sends incremental kWh values via webhook every 10mins to HA. Works ok, numbers are very same as FPL says, but more detailed and properly distributed over a time. (minimal amount for FPL is 1kWh, so if you have continuous consumption let’s say 200W, FPL will show for 4 hours zero and fifth hour would be 1kWh). Of course, this Vue thing it is a cloud solution, so it has its bad sides Next logical step for me would be flash my Vue with ESP firmware and use ESPhome. But ESP is something I don’t have experience at all, so step by step ;) Anyways, this doesn’t belong to this thread. Sorry.

M.
